net/sched/act_skbedit.c

```diff
@@ -50,6 +53,30 @@ static int tcf_skbedit(struct sk_buff *skb, const struct tc_action *a,
 	}
 	if (d->flags & SKBEDIT_F_PTYPE)
 		skb->pkt_type = d->ptype;
+	if (d->flags & SKBEDIT_F_INHERITDSFIELD) {
+		int wlen = skb_network_offset(skb);
+
+		switch (tc_skb_protocol(skb)) {
+		case htons(ETH_P_IP):
+			wlen += sizeof(struct iphdr);
+			if (!pskb_may_pull(skb, wlen)) {
+				spin_unlock(&d->tcf_lock);
+				return TC_ACT_SHOT;
```

Comment:
I think it's better to create a label `err:` at the end of the function and `goto err;` here.